ast/src/analyzed/display.rs

```diff
@@ -378,3 +437,79 @@ impl Display for PolynomialReference {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::iter::once;
+
+    use powdr_number::GoldilocksField;
+    use powdr_pil_analyzer::analyze_string;
+    use pretty_assertions::assert_eq;
+    use test_log::test;
+
+    type TestCase<'a> = (&'a str, &'a str);
+
+    fn test_paren(to_declare: &[&str], (input, expected): &TestCase) {
+        // Display on `Analysis` pads each line by 4 spaces
+        let padding: String = " ".repeat(4);
+
+        // Introduce witness columns for the referenced variables
+        let declarations = to_declare
+            .iter()
+            .map(|to_declare| format!("col witness {to_declare};"));
+
+        // Wrap the expression we're testing in a polynomial identity
+        let wrap = |e| {
+            declarations
+                .clone()
+                .chain(once(format!("{e} = 0;")))
+                .collect::<Vec<_>>()
+                .join(&format!("\n{padding}"))
+        };
+
+        let (input, expected) = &(&wrap(input), &wrap(expected));
+        let analyzed = analyze_string::<GoldilocksField>(input);
+        let printed = analyzed.to_string();
+
+        assert_eq!(expected.trim(), printed.trim());
+    }
+
+    #[test]
+    fn test_binary_op_parentheses() {
+        let test_cases: Vec<TestCase> = vec![
```

Comment:
Can you add a test that shows associativity of `**` and `*` is handled correctly?